usng4j-impl/src/main/java/org/codice/usng4j/impl/UsngCoordinateImpl.java

```diff
@@ -183,20 +207,44 @@ private static UsngCoordinate parseCoordinateString(
       throw new ParseException(message, 0);
     }
 
+    int captureGroupsCount = m.groupCount();
     int zoneNumber = Integer.parseInt(m.group(1));
     char latitudeBandLetter = m.group(2).toCharArray()[0];
 
     if (m.group(3) != null) {
       char columnLetter = m.group(3).toCharArray()[0];
       char rowLetter = m.group(3).toCharArray()[1];
 
-      if (!(StringUtils.isEmpty(m.group(4)) || StringUtils.isEmpty(m.group(5)))) {
-        int easting = Integer.parseInt(m.group(4).trim());
-        int northing = Integer.parseInt(m.group(5).trim());
+      if (captureGroupsCount > 4
```

Comment:
There's lot of 4's and 5's and magic numbers. Would it be possible to store these as constant int's to make the code more readable?

Reply:
I removed some of the code referencing those numbers and added some comments explaining what they mean.